numpyro/distributions/continuous.py

```diff
@@ -1680,38 +1680,46 @@ def infer_shapes(df, loc, scale_tril):
         return batch_shape, event_shape
 
 
-class MultivariateBeta(Distribution):
+class GaussianCopula(Distribution):
     arg_constraints = {
-        "concentration1": constraints.positive,
-        "concentration0": constraints.positive,
+        "covariance_matrix": constraints.positive_definite,
+        "precision_matrix": constraints.positive_definite,
+        "scale_tril": constraints.lower_cholesky,
     }
-    reparametrized_params = ["concentration1", "concentration0"]
+    support = constraints.real_vector
+    reparametrized_params = [
+        "covariance_matrix",
+        "precision_matrix",
+        "scale_tril",
+    ]
+
     support = constraints.unit_interval
 
     def __init__(
         self,
-        mv_normal,
-        concentration1,
-        concentration0,
+        transform_dist,
+        covariance_matrix=None,
+        precision_matrix=None,
+        scale_tril=None,
         validate_args=None,
     ):
-        self.concentration1, self.concentration0 = promote_shapes(
-            concentration1, concentration0
+        self.transform_dist = transform_dist
+        self.base_dist = MultivariateNormal(
+            covariance_matrix=covariance_matrix,
+            precision_matrix=precision_matrix,
+            scale_tril=scale_tril,
         )
+        self.normal = Normal(scale=jnp.diag(self.base_dist.covariance_matrix))
 
-        self.beta = Beta(self.concentration1, self.concentration0)
-        self.normal = Normal(scale=jnp.diag(mv_normal.covariance_matrix))
-        self.mv_normal = mv_normal
+        event_shape = self.base_dist.event_shape
+        delta_dim = len(event_shape) - len(transform_dist.event_shape)
 
         batch_shape = lax.broadcast_shapes(
-            jnp.shape(concentration1),
-            jnp.shape(concentration0),
-            mv_normal.batch_shape,
+            self.transform_dist.batch_shape[:-delta_dim],
+            self.base_dist.batch_shape,
```

Comment:
I guess batch shape is `lax.broadcast_shapes(transform_dist.batch_shape, base_dist.batch_shape[:-1])` if `len(base_dist.batch_shape) >= 1` else `transform_dist.batch_shape` (so that we can specify `Beta(1, 1), correlation_matrix=jnp.eye(3)` as arguments)